infra_event_notifier/notifier.py

```diff
@@ -0,0 +1,55 @@
+from typing import Mapping
+
+from backends.datadog import send_event
+from backends.jira import create_issue
+from backends.slack import send_notification
+
+
+class Notifier:
+    def __init__(
+        self,
+        datadog_api_key: str = None,
+        slack_api_key: str = None,
+        jira_api_key: str = None,
+    ) -> None:
+        self.datadog_api_key = datadog_api_key
+        self.slack_api_key = slack_api_key
+        self.jira_api_key = jira_api_key
+
+    def send_datadog_event(
+        self,
+        title: str,
+        text: str,
+        tags: Mapping[str, str],
+        alert_type: str,
+    ):
+        """
+        Sends an event to Datadog.
+
+        :param title: Title of DD event
+        :param text: Body of event
+        :param tags: dict storing event tags
+        :param alert_type: Type of event if using an event monitor, see https://docs.datadoghq.com/api/latest/events/
+        """
+        if not self.datadog_api_key:
+            raise ValueError("datadog_api_key must be set to send events")
```

Comment:
I don't see the reason for the three method.
You are already telling this class which Notifications to use by setting some tokens.
Why not having a single `send` method that send the notification on all the channels we have a token for and skips those we do not have a token for ?

The only difference in the interface are the tags for slack you can ignore.

Reply:
I'll switch the structure to this, but my thinking was that the parameters for slack/jira will diverge more as they're written (jira needs project ID, slack needs channel, etc)